File: Dr.NutrizioNino.Models/Dto/DailySimulationDto.cs
namespace Dr.NutrizioNino.Models.Dto;

// ── Richieste ──────────────────────────────────────────────

public record CreateDailySimulationDto(string Name);

public record RenameDailySimulationDto(string Name);

/// <summary>SourceType: 0=Food 1=Dish</summary>
public record AddSimulationEntryDto(Guid SectionId, byte SourceType, Guid SourceId, decimal QuantityGrams);

public record UpdateEntryQuantityDto(decimal QuantityGrams);

// ── Risposte ───────────────────────────────────────────────

public record DailySimulationListItemDto(Guid Id, string Name, DateTime CreatedAt, int EntryCount);

public record DailySimulationEntryNutrientDto(
    string Name,
    int PositionOrder,
    decimal Quantity,
    string UnitAbbreviation
);

public record DailySimulationEntryDto(
    Guid Id,
    string SourceName,
    byte SourceType,
    decimal QuantityGrams,
    IList<DailySimulationEntryNutrientDto> Nutrients
);

public record DailySimulationSectionDto(
    Guid SectionId,
    string SectionName,
    IList<DailySimulationEntryDto> Entries
);

public record DailySimulationDetailDto(
    Guid Id,
    string Name,
    DateTime CreatedAt,
    IList<DailySimulationSectionDto> Sections
);

// ── Confronto ─────────────────────────────────────────────

public record SimulationCompareNutrientDto(
    string Name,
    int PositionOrder,
    decimal? Sim1Quantity,
    decimal? Sim2Quantity,
    string UnitAbbreviation
);

public record DailySimulationCompareDto(
    Guid Sim1Id,
    string Sim1Name,
    Guid Sim2Id,
    string Sim2Name,
    IList<SimulationCompareNutrientDto> Nutrients
);

File: src/Dr.NutrizioNino.WebVue/src/Interfaces/dailySimulations/DailySimulationDto.ts
export interface DailySimulationListItemDto {
  id: string
  name: string
  createdAt: string
  entryCount: number
}

export interface DailySimulationEntryNutrientDto {
  name: string
  positionOrder: number
  quantity: number
  unitAbbreviation: string
}

export interface DailySimulationEntryDto {
  id: string
  sourceName: string
  sourceType: number // 0=Food 1=Dish
  quantityGrams: number
  nutrients: DailySimulationEntryNutrientDto[]
}

export interface DailySimulationSectionDto {
  sectionId: string
  sectionName: string
  entries: DailySimulationEntryDto[]
}

export interface DailySimulationDetailDto {
  id: string
  name: string
  createdAt: string
  sections: DailySimulationSectionDto[]
}

export interface SimulationCompareNutrientDto {
  name: string
  positionOrder: number
  sim1Quantity: number | null
  sim2Quantity: number | null
  unitAbbreviation: string
}

export interface DailySimulationCompareDto {
  sim1Id: string
  sim1Name: string
  sim2Id: string
  sim2Name: string
  nutrients: SimulationCompareNutrientDto[]
}

// ── Richieste ──────────────────────────────────────────────

export interface AddSimulationEntryRequest {
  sectionId: string
  sourceType: number
  sourceId: string
  quantityGrams: number
}
